src/load-configuration-file.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import fs from "fs";
+import {promisify} from "util";
+import {version} from "../package.json";
+import JsonSchema, {DETAILED} from "@hyperjump/json-schema";
+
+import type {ILog, Options} from "./types.js";
+import FileReferenceLogger from "./file-reference-logger.js";
+import checksyncSchema from "./checksync.schema.json";
+
+const readFileAsync = promisify(fs.readFile);
+
+export default async function loadConfigurationFile(
+    configPath: string,
+    log: ILog,
+): Promise<?Options> {
+    log.verbose(() => `Loading configuration from ${configPath}`);
+    const rcLog = new FileReferenceLogger(configPath, log);
+    try {
+        // Load the configuration JSON.
+        const rcFile: string = await readFileAsync(configPath);
+        const rcJson = JSON.parse(rcFile);
+
+        const rcForValidation = {...rcJson};
+        if (!rcForValidation.$schema) {
+            rcForValidation.$schema = "file://./checksync.schema.json";
+        }
+        if (!rcForValidation.$schemaVersion) {
+            rcForValidation.$schemaVersion = version;
+        }
+        JsonSchema.setShouldMetaValidate(false);
+
+        // Load the schema versions into the schema validator, giving it a URI to ID it.
+        JsonSchema.add(
+            checksyncSchema,
+            "file://./checksync.schema.json",
+            version,
+        );
+
+        // Now get that schema using the same URI, and use it to validate the
+        // configuration we loaded.
+        const schema = await JsonSchema.get("file://./checksync.schema.json");
+
+        log.verbose(() => `Validating configuration`);
+        const validation = await JsonSchema.validate(
```

Comment:
The validation does include details of the actual errors, but it requires some work to turn that into human readable info that can be acted upon, so I'm going to not do that for now.